formats/protobuf/commonMain/src/kotlinx/serialization/protobuf/internal/Streams.kt

```diff
@@ -189,42 +200,33 @@ internal class ByteArrayOutput {
     }
 
     fun encodeVarint32(value: Int) {
-        ensureCapacity(5)
         // Fast-path: unrolled loop for single byte
+        ensureCapacity(5)
```

Comment:
Why now for `long` we ensure dynamic length but here always 5?

Reply:
Because 32-bit varints are used on a regular basis for tags, it's always beneficial to allocate all the space.
Also, because tags are small, one-byte fast-path is used and we do not have to calculate varint length at all for such values